tests/contract/test_resource_client.py

```diff
@@ -161,6 +162,19 @@ def test_prune_properties_from_model():
     assert document == {"one": "two", "array": ["first"]}
 
 
+def test_prune_properties_except_identifier():
+    model = {
+        "foo": "bar",
+        "spam": "eggs",
+        "one": "two",
+        "array": ["first", "second"],
+    }
+    prune_properties_except_identifier(
+        model, [("properties", "foo"), ("properties", "spam")]
+    )
+    assert model == {"foo": "bar", "spam": "eggs"}
```

Comment:
can we add a test for a nested property?

Reply:
makes sense.